openstack/identity/v3/applicationcredentials/requests.go

```diff
@@ -0,0 +1,95 @@
+package applicationcredentials
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// ListOptsBuilder allows extensions to add additional parameters to
+// the List request
+type ListOptsBuilder interface {
+	ToApplicationCredentialListQuery() (string, error)
+}
+
+// ListOpts provides options to filter the List results.
+type ListOpts struct {
+	// Name filters the response by an application credential name
+	Name string `json:"name"`
+}
+
+// ToApplicationCredentialListQuery formats a ListOpts into a query string.
+func (opts ListOpts) ToApplicationCredentialListQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+// List enumerates the ApplicationCredentials to which the current token has access.
+func List(client *gophercloud.ServiceClient, userID string, opts ListOptsBuilder) pagination.Pager {
+	url := listURL(client, userID)
+	if opts != nil {
+		query, err := opts.ToApplicationCredentialListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		return ApplicationCredentialPage{pagination.LinkedPageBase{PageResult: r}}
+	})
+}
+
+// Get retrieves details on a single user, by ID.
+func Get(client *gophercloud.ServiceClient, userID string, id string) (r GetResult) {
+	_, r.Err = client.Get(getURL(client, userID, id), &r.Body, nil)
+	return
+}
+
+// CreateOptsBuilder allows extensions to add additional parameters to
+// the Create request.
+type CreateOptsBuilder interface {
+	ToApplicationCredentialCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts provides options used to create an application credential.
+type CreateOpts struct {
+	// The name of the application credential.
+	Name string `json:"name,omitempty"`
```

Comment:
Looks like this is required? https://github.com/openstack/keystone/blob/ef87a202127f0ac84b857716e72595ce75058378/keystone/application_credential/schema.py#L48

Omitting the name in `CreateOpts` yields an error, too.

Reply:
I've never seen, that the `required:true` tag worked within go. It is more likely cosmetic change.
Done.

Reply:
It's a Gophercloud-specific tag: https://github.com/gophercloud/gophercloud/blob/master/params.go#L64